pkg/controller/migmigration/task.go

```diff
@@ -1194,6 +1207,46 @@ func (t *Task) anyFlags(phase Phase) (bool, error) {
 	return phase.any == uint16(0), nil
 }
 
+// Evaluate `any` flags overall to look for artifacts (not per phase).
+// 1. Check if the `StagePodsCreated` condition has been created either due to pod migration from templates or running pods of application.
+// 2. Check if there are any PVs and image stream that were migrated.
+// 3. Check for any pod templates that were present in the source cluster namespace which could have been migrated.
+// 4. If none of the above cases have been encountered, then the namespace is likely to be an empty one.
+func (t *Task) anyArtifacts() (bool, error) {
+	if t.Owner.Status.HasCondition(StagePodsCreated) || t.hasVerify() {
+		return true, nil
+	}
+
+	if t.directVolumeMigration() {
+		return true, nil
+	}
+
+	anyPVs, moveSnapshotPVs := t.hasPVs()
+
+	anyImageStream, err := t.hasImageStreams()
+	if err != nil {
+		return false, liberr.Wrap(err)
+	}
+
+	if t.hasStageBackup(anyImageStream, anyPVs, moveSnapshotPVs) {
+		return true, nil
+	}
+
+	client, err := t.getSourceClient()
+	if err != nil {
+		return false, liberr.Wrap(err)
+	}
+
+	podTemplates, err := migpods.ListTemplatePods(client, t.sourceNamespaces())
```

Comment:
Would this report empty namespaces for namespaces with deployments that are scaled down to zero?